rts/Lua/LuaUnsyncedCtrl.cpp

```diff
@@ -851,13 +851,21 @@ int LuaUnsyncedCtrl::SetCameraState(lua_State* L)
 	if (mouse == nullptr)
 		return 0;
 
-	if (!lua_istable(L, 1))
-		luaL_error(L, "[%s(stateTable[, camTransTime[, transTimeFactor[, transTimeExpon] ] ])] incorrect arguments", __func__);
+	const bool hasState = lua_istable(L, 1);
+
+	if (!(hasState || lua_isnil(L, 1)))
+		luaL_error(L, "[%s([ stateTable[, camTransTime[, transTimeFactor[, transTimeExpon] ] ] ])] incorrect arguments", __func__);
 
 	camHandler->SetTransitionParams(luaL_optfloat(L, 3, camHandler->GetTransitionTimeFactor()), luaL_optfloat(L, 4, camHandler->GetTransitionTimeExponent()));
 	camHandler->CameraTransition(luaL_optfloat(L, 2, 0.0f));
 
-	const bool retval = camHandler->SetState(ParseCamStateMap(L, 1));
+	CCameraController::StateMap camState;
+	if (hasState)
+		camState = ParseCamStateMap(L, 1);
+	else
+		camHandler->GetState(camState);
+
+	const bool retval = camHandler->SetState(camState);
```

Comment:
IMO it hurts readability. 
Also I was going to change SetCameraState() implementation because of https://github.com/beyond-all-reason/spring/issues/16

TODO if you insist.

Reply:
In this case yes, it collapses 5 lines to 1 and avoids a copy / no-op `SetState` call.

Reply:
I'm not sure SetState(GetState()) is no-op. Besides assigning params (no-op) it also calls Update(). We will lose Update() if we collapse it like that.

Reply:
Right, forgot `SetState` also called `Update` along the way. In that case I'd rather have `const bool retval = camHandler->SetState(hasState? ParseCamStateMap(L, 1): camHandler->GetState());` where `GetState()` is simply:

```
CCameraController::StateMap GetState() const {
    CCameraController::StateMap sm;
    GetState(sm);
    return sm;
}
```